api/dotnet/src/lib.rs

```diff
@@ -0,0 +1,596 @@
+            }
+            Type::Int32 | Type::Float32 => {
+                val_type = DotNetType::NUMBER;
+            }
+            Type::Bool => {
+                val_type = DotNetType::BOOL;
+            }
+            Type::Image => {
+                val_type = DotNetType::IMAGE;
+            }
+            Type::Struct { fields, .. } => {
+                val_type = DotNetType::STRUCT;
+                val_struct = true;
+
+                for (field, s_type) in &fields {
+                    let sval_type;
+
+                    match s_type {
+                        Type::String => sval_type = DotNetType::STRING,
+                        Type::Int32 | Type::Float32 => sval_type = DotNetType::NUMBER,
+                        Type::Bool => sval_type = DotNetType::BOOL,
+                        Type::Image => sval_type = DotNetType::IMAGE,
+                        Type::Struct { .. } => {
+                            panic!("struct inside struct not supported");
+                        }
+                        _ => {
+                            panic!("Slint type not supported inside a struct");
+                        }
+                    }
+
+                    val_props.push(DotNetValue {
+                        type_name: field.to_string(),
+                        type_type: sval_type as i32,
+                        type_value: "".to_string(),
+                        is_struct: false,
+                        struct_props: Vec::new(),
+                    });
+                }
+            }
+            Type::Callback { .. } => {
+                // FIX-ME: when we want to implement callback with arguments
+                // and return types we have to change this
+                continue;
+            }
+            _ => {
+                panic!("Slint type not supported");
+            }
+        }
+
+        m_props.push(DotNetValue {
+            type_name: p_name,
+            type_type: val_type as i32,
+            type_value: val_val,
+            is_struct: val_struct,
+            struct_props: val_props,
+        });
+    }
+
+    let m_calls = ret_handle.callbacks().collect();
+
+    let tokens = Tokens { props: m_props, calls: m_calls };
+
+    tokens
+}
+
+#[net]
+pub fn create(path: &str) {
+    printdebug!("create()");
+
+    let mut compiler = slint_interpreter::ComponentCompiler::default();
+    let path = std::path::Path::new(path);
+    let ret_handle = async_std::task::block_on(compiler.build_from_path(path)).unwrap();
+
+    slint_interpreter::print_diagnostics(compiler.diagnostics());
+    let component = ret_handle.create().unwrap();
+
+    CURRENT_INSTANCE.with(|current| current.replace(Some(component.clone_strong())));
+
+    TIMER_POOL.with(|pool| {
+        pool.replace(Some(Vec::new()));
+    });
+}
+
+#[net]
+pub fn get_properties() -> Vec<DotNetValue> {
+    printdebug!("get_properties()");
+
+    let mut ret: Vec<DotNetValue> = Vec::new();
+
+    CURRENT_INSTANCE.with(|current| {
```

Comment:
The use of thread locals is something I find worth improving. It would be better if the instance could be wrapped in an opaque type and become a parameter to all the functions (and return value for the initial loading).

As far as I can see rnet support Box and Arc wrapped types. Is this something you’d like to pursue?